packages/core/src/shared/awsClientBuilderV3.ts

```diff
@@ -77,6 +78,25 @@ export class AWSClientBuilderV3 {
         return shim
     }
 
+    private keyAwsService<C extends AwsClient>(
+        type: AwsClientConstructor<C>,
+        options?: Partial<AwsClientOptions>,
+        region?: string,
+        userAgent: boolean = true,
+        settings?: DevSettings
+    ): string {
+        // Serializing certain objects in the args allows us to detect when nested objects change (ex. new retry strategy, endpoints)
+        return [
+            String(type),
+            JSON.stringify(options),
+            region,
+            userAgent ? '1' : '0',
+            settings ? JSON.stringify(settings.get('endpoints', {})) : '',
+        ].join(':')
+    }
+
+    public getAwsService = memoizeWith(this.createAwsService.bind(this), this.keyAwsService.bind(this))
```

Comment:
memoize may not be gaining much for this use-case. 

because this is an important core module, we may want more visibility into the long-lived container that stores the clients. this module could store them in a map, that could be inspected and/or logged.

Reply:
That makes sense, this will make it easier to add custom-logic later.